Week3/Homework/Js-exercises/cardValidator.js

```diff
@@ -0,0 +1,45 @@
+"use strict";
+
+const cardValidator = function(cardNumber) {
+  //first converts our number into an array of strings then using for of, converts the string array to numbers array
+  if (Number.isInteger(Number(cardNumber)) !== true) {
+    return console.log("invalid characters!");
+  }
+  let strArr = cardNumber.toString().split("");
+  const numArr = [];
+  for (let ele of strArr) {
+    numArr.push(Number(ele));
+  }
+  console.log(numArr.length);
+  if (numArr.length !== 16) {
+    return console.log("invalid card ,there is less or more than 16 digits!");
+  }
+  //pushes the numbers with -1 index(numbers who dont exist in indOfArr array) so if length.indOfArr is less than or equal to 1 it means that all numbers are same
+  const indOfArr = [];
+  for (let index = 0; index < numArr.length; index++) {
+    const number = numArr[index];
+    if (indOfArr.indexOf(number) === -1) {
+      indOfArr.push(number);
+    }
+  }
+  if (indOfArr.length <= 1) {
+    return console.log("invalid card, all digits can't be same");
+  }
```

Comment:
Ah. Instead of writing so much code for this logic, I would suggest that you look into a new concept in JavaScript called **Sets** and see if it helps you to write lesser code - https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Set